src/bp/core/services/stats-service.ts

```diff
@@ -207,4 +263,67 @@ export class StatsService {
   private async getCollaboratorsCount(): Promise<number> {
     return (await this.authService.getAllUsers()).length
   }
+
+  private modulesWhitelist = {
+    builtin: true,
+    analytics: true,
+    'basic-skills': true,
+    'channel-web': true
+  }
+
+  private async getBuiltinActionsStats() {
+    return {
+      timestamp: new Date().toISOString(),
+      uuid: uuid.v4(),
+      schema: '1.0.0',
+      source: 'server',
+      server: await this.getServerStats(),
+      event_type: 'builtin_actions',
+      event_data: { schema: '1.0.0', flows: await this.getFlows() }
+    }
+  }
+
+  private async getFlows() {
+    const flows = await this.ghostService.bots().directoryListing('/', '*/flows/*.flow.json')
+
+    const parsedFlows = await Promise.all(
+      flows.map(async element => {
+        const { dir, base: flowName } = path.parse(element)
+        const actions = (await this.ghostService.bots().readFileAsObject<any>(dir, flowName)).nodes
+          .map(node => this.getActionsFromNode(node))
+          .reduce((acc, cur) => acc.concat(cur))
+          .filter(action => BUILTIN_MODULES.includes(action[0].split('/')[0]))
+
+        const botID = dir.split('/')[0]
+
+        return { flowName, botID, actions }
+      })
+    )
+      .filter(flow => flow.actions.length > 0)
+      .map(flow => this.parseFlow(flow))
+
+    return parsedFlows
+  }
+
+  private getActionsFromNode(node) {
+    const onEnter = node.onEnter ? node.onEnter.map(action => action.split(' ')) : []
+    const onReceive = node.onReceive ? node.onReceive.map(action => action.split(' ')) : []
+    return onEnter.concat(onReceive)
```

Comment:
In pretty much all of botpress code we use spread operator syntax instead of .concat
```
return [
  ...(node.onEnter || []).map(action => action.split(' ')),
  ...(node. onReceive || []).map(action => action.split(' '))
]
```